database.cc

```diff
@@ -2321,6 +2321,69 @@ future<> database::drain() {
     co_await _commitlog->shutdown();
 }
 
+static sstring extract_cf_name(const sstring& directory_name) {
+    // cf directory is of the form: 'cf_name-timestamp'
+    return directory_name.substr(0, directory_name.find("-"));
+}
+
+future<std::unordered_map<sstring, std::unordered_map<std::pair<sstring, sstring>, db::snapshot_ctl::snapshot_details, utils::tuple_hash>>> database::get_snapshot_details() {
```

Comment:
As we talked on slack: this could return a simple vector of structs, you never use the `unordered_map` structure.